types/currency.go

```diff
@@ -28,34 +48,27 @@ func NewCurrency64(x uint64) (c Currency) {
 	return
 }
 
+// Add returns a new Currency value y = c + x.
+func (c Currency) Add(x Currency) (y Currency) {
+	y.i.Add(&c.i, &x.i)
+	return
+}
+
 // Big returns the value of c as a *big.Int. Importantly, it does not provide
-// access to the c's internal big.Int object, only a copy. This is in
-// accordance with the immutability constraint described above.
+// access to the c's internal big.Int object, only a copy.
 func (c Currency) Big() *big.Int {
 	return new(big.Int).Set(&c.i)
 }
 
 // Cmp compares two Currency values. The return value follows the convention
-// of the math/big package.
+// of math/big.
 func (c Currency) Cmp(y Currency) int {
 	return c.i.Cmp(&y.i)
 }
 
-// Sign returns the sign of a Currency. The return value follows of the
-// convention of the math/big package.
-func (c Currency) Sign() int {
-	return c.i.Sign()
-}
-
-// Add returns a new Currency value y = c + x.
-func (c Currency) Add(x Currency) (y Currency) {
-	y.i.Add(&c.i, &x.i)
-	return
-}
-
-// Sub returns a new Currency value y = c - x.
-func (c Currency) Sub(x Currency) (y Currency) {
-	y.i.Sub(&c.i, &x.i)
+// Div returns a new Currency value y = c / x.
+func (c Currency) Div(x Currency) (y Currency) {
+	y.i.Div(&c.i, &x.i)
```

Comment:
life is hard.

I left Marshal and Unmarshal together because it seemed really natural. New is at the top.